x/stake/tick_test.go

```diff
@@ -120,14 +141,264 @@ func TestProcessProvisions(t *testing.T) {
 	assert.Equal(t, initialUnbonded, pool.UnbondedPool)
 	//panic(fmt.Sprintf("debug total %v, bonded  %v, diff %v\n", p.TotalSupply, p.BondedPool, pool.TotalSupply-pool.BondedPool))
 
+	calculatedTotalSupply := totalSupply + provisionTallied
+	calculatedBondedSupply := bondedShares + provisionTallied
```

Comment:
`TotalSupply` and `provisionsTallied` have the same units (bond-denom), whereas `bondedShares` most certainly have a different unit (shares). You cannot add terms of different units, so whatever you're calculating in `calculatedBondedSupply` is bit of a black box.

Reply:
Also with the rational type we need to add like this: `RatA = RatA.Add(RatB)

Reply:
yes totally makes sense. i just read through the staking spec and I see now that shares aren't the same as the provisions. will fix

Reply:
fixed